ddtrace/tracer/tracer_test.go

```diff
@@ -44,8 +45,27 @@ func (t *tracer) newChildSpan(name string, parent *span) *span {
 	return t.StartSpan(name, ChildOf(parent.Context())).(*span)
 }
 
+var (
+	// timeMultiplicator specifies by how long to extend waiting times.
+	// It may be altered in some envinronment (like AppSec) where things
+	// move slower and could otherwise create flaky tests.
+	timeMultiplicator = time.Duration(1)
+
+	// integration indicates if the test suite should run integration tests.
+	integration bool
+)
+
+func TestMain(m *testing.M) {
+	if maininternal.BoolEnv("DD_APPSEC_ENABLED", false) {
+		// things are slower with AppSec; double wait times
+		timeMultiplicator = time.Duration(2)
```

Comment:
Why not simply make this the default?

Reply:
Well, since we know that no multiplication is needed without AppSec, and that those tests would consistently pass, I was thinking it might be better to not change it for those cases. If they suddenly start failing for some reason, that would potentially identify a performance regression which would likely be good to know about. Arguably the same thing happens with AppSec on.

Reply:
I've seen those tests fail without AppSec before, albeit at a lower rate. I think this PR will lower the overall suite failure rate, but I suspect we'll continue to see these tests to flake occasionally.

Reply:
We can do as you propose. I don't mind.